### ISSUE TITLE
Using a camera initiate a crash.

### ISSUE THREAD

**Issue by christianclavet** [BearishSun/BansheeEngine#117]
Build date: (December 16, 2017) (Windows 10 64bit, OpenGL)

Problem description:
Using the scene camera work well, but creating a user camera in the editor will crash the application.

How to reproduce:
1. Open the application and add some assets to the scene
2. Create a camera (no crash here)
3. Click on the GAME TAB to display the camera rendering
4. The engine should crash

Here is the debug output from the MSVC 2015 debugger:
```
A fatal error occurred and the program has to terminate!
  - Error: RenderingAPIException
  - Description: OpenGL error: GL_INVALID_VALUE error generated. The number of work groups exceeds implementation limits in at least one dimension.
  - In function: void __cdecl bs::ct::openGlErrorCallback(unsigned int,unsigned int,unsigned int,unsigned int,int,const char *,void *)
  - In file: c:\users\public\projets\bansheeengine\source\bansheeglrenderapi\bsglrenderapi.cpp:2829

Stack trace: 
bs::CrashHandler::reportCrash() - 0x140710194603333 File[bswin32crashhandler.cpp:466 (0)] Module[BansheeUtility.dll]
bs::ct::openGlErrorCallback() - 0x140710072221163 File[bsglrenderapi.cpp:2829 (181)] Module[BansheeGLRenderAPI.dll]
0x1697302123 Module[nvoglv64.DLL]
0x1694946511 Module[nvoglv64.DLL]
0x1694946748 Module[nvoglv64.DLL]
0x1694946661 Module[nvoglv64.DLL]
0x1694901362 Module[nvoglv64.DLL]
<lambda_5e8475dca5a4d0f691b093fff5004e9b>::operator()() - 0x140710072133337 File[bsglrenderapi.cpp:1287 (0)] Module[BansheeGLRenderAPI.dll]
bs::ct::GLRenderAPI::dispatchCompute() - 0x140710072206587 File[bsglrenderapi.cpp:1291 (0)] Module[BansheeGLRenderAPI.dll]
bs::ct::LightGridLLCreationMat::execute() - 0x140709894586796 File[bslightgrid.cpp:125 (73)] Module[RenderBeast.dll]
bs::ct::LightGrid::updateGrid() - 0x140709894594070 File[bslightgrid.cpp:302 (0)] Module[RenderBeast.dll]
bs::ct::RendererView::updateLightGrid() - 0x140709894562665 File[bsrendererview.cpp:525 (0)] Module[RenderBeast.dll]
bs::ct::RendererViewGroup::determineVisibility() - 0x140709894551542 File[bsrendererview.cpp:625 (0)] Module[RenderBeast.dll]
bs::ct::RenderBeast::renderAllCore() - 0x140709894242265 File[bsrenderbeast.cpp:324 (0)] Module[RenderBeast.dll]
std::_Invoker_pmf_pointer::_Call<void (__cdecl bs::ct::RenderBeast::*)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 & __ptr64,bs::ct::FrameTimings & __ptr64>() - 0x140709893917617 File[type_traits:1342 (0)] Module[RenderBeast.dll]
std::invoke<void (__cdecl bs::ct::RenderBeast::*& __ptr64)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 & __ptr64,bs::ct::FrameTimings & __ptr64>() - 0x140709893997266 File[type_traits:1445 (0)] Module[RenderBeast.dll]
std::_Invoke_ret<void (__cdecl bs::ct::RenderBeast::*& __ptr64)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 & __ptr64,bs::ct::FrameTimings & __ptr64>() - 0x140709893940582 File[type_traits:1476 (0)] Module[RenderBeast.dll]
std::_Call_binder<std::_Unforced,0,1,void (__cdecl bs::ct::RenderBeast::*)(bs::ct::FrameTimings) __ptr64,std::tuple<bs::ct::RenderBeast * __ptr64,bs::ct::FrameTimings>,std::tuple<> >() - 0x140709893918224 File[functional:827 (0)] Module[RenderBeast.dll]
std::_Binder<std::_Unforced,void (__cdecl bs::ct::RenderBeast::*)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 const,bs::ct::FrameTimings & __ptr64>::operator()<>() - 0x140709893916376 File[functional:881 (136)] Module[RenderBeast.dll]
std::_Invoker_functor::_Call<std::_Binder<std::_Unforced,void (__cdecl bs::ct::RenderBeast::*)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 const,bs::ct::FrameTimings & __ptr64> & __ptr64>() - 0x140709893917230 File[type_traits:1377 (0)] Module[RenderBeast.dll]
std::invoke<std::_Binder<std::_Unforced,void (__cdecl bs::ct::RenderBeast::*)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 const,bs::ct::FrameTimings & __ptr64> & __ptr64>() - 0x140709893997614 File[type_traits:1445 (0)] Module[RenderBeast.dll]
std::_Invoke_ret<void,std::_Binder<std::_Unforced,void (__cdecl bs::ct::RenderBeast::*)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 const,bs::ct::FrameTimings & __ptr64> & __ptr64>() - 0x140709893940946 File[type_traits:1462 (0)] Module[RenderBeast.dll]
std::_Func_impl<std::_Binder<std::_Unforced,void (__cdecl bs::ct::RenderBeast::*)(bs::ct::FrameTimings) __ptr64,bs::ct::RenderBeast * __ptr64 const,bs::ct::FrameTimings & __ptr64>,std::allocator<int>,void>::_Do_call() - 0x140709894117205 File[functional:214 (0)] Module[RenderBeast.dll]
std::_Func_class<void>::operator()() - 0x140709846421162 File[functional:280 (0)] Module[BansheeCore.dll]
bs::CommandQueueBase::playbackWithNotify() - 0x140709846476825 File[bscommandqueue.cpp:127 (0)] Module[BansheeCore.dll]
bs::CommandQueueBase::playback() - 0x140709846476143 File[bscommandqueue.cpp:141 (0)] Module[BansheeCore.dll]
std::_Invoker_pmf_pointer::_Call<void (__cdecl bs::CommandQueueBase::*)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64>() - 0x140709847041510 File[type_traits:1342 (0)] Module[BansheeCore.dll]
std::invoke<void (__cdecl bs::CommandQueueBase::*& __ptr64)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64>() - 0x140709847046754 File[type_traits:1445 (0)] Module[BansheeCore.dll]
std::_Invoke_ret<void (__cdecl bs::CommandQueueBase::*& __ptr64)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64>() - 0x140709847042566 File[type_traits:1476 (0)] Module[BansheeCore.dll]
std::_Call_binder<std::_Unforced,0,1,void (__cdecl bs::CommandQueueBase::*)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,std::tuple<bs::CommandQueueBase * __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64>,std::tuple<> >() - 0x140709847041744 File[functional:827 (0)] Module[BansheeCore.dll]
std::_Binder<std::_Unforced,void (__cdecl bs::CommandQueueBase::*)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64>::operator()<>() - 0x140709847041256 File[functional:881 (136)] Module[BansheeCore.dll]
std::_Invoker_functor::_Call<std::_Binder<std::_Unforced,void (__cdecl bs::CommandQueueBase::*)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64> & __ptr64>() - 0x140709847041374 File[type_traits:1377 (0)] Module[BansheeCore.dll]
std::invoke<std::_Binder<std::_Unforced,void (__cdecl bs::CommandQueueBase::*)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64> & __ptr64>() - 0x140709847046862 File[type_traits:1445 (0)] Module[BansheeCore.dll]
std::_Invoke_ret<void,std::_Binder<std::_Unforced,void (__cdecl bs::CommandQueueBase::*)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64> & __ptr64>() - 0x140709847042690 File[type_traits:1462 (0)] Module[BansheeCore.dll]
std::_Func_impl<std::_Binder<std::_Unforced,void (__cdecl bs::CommandQueueBase::*)(std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64) __ptr64,bs::CommandQueueBase * __ptr64 & __ptr64,std::queue<bs::QueuedCommand,std::deque<bs::QueuedCommand,bs::StdAlloc<bs::QueuedCommand,bs::GenAlloc> > > * __ptr64 & __ptr64>,std::allocator<int>,void>::_Do_call() - 0x140709847049909 File[functional:214 (0)] Module[BansheeCore.dll]
std::_Func_class<void>::operator()() - 0x140709846421162 File[functional:280 (0)] Module[BansheeCore.dll]
bs::CommandQueueBase::playbackWithNotify() - 0x140709846476825 File[bscommandqueue.cpp:127 (0)] Module[BansheeCore.dll]
bs::CoreThread::runCoreThread() - 0x140709847033156 File[bscorethread.cpp:119 (0)] Module[BansheeCore.dll]
std::_Invoker_pmf_pointer::_Call<void (__cdecl bs::CoreThread::*)(void) __ptr64,bs::CoreThread * __ptr64 & __ptr64>() - 0x140709846874372 File[type_traits:1342 (0)] Module[BansheeCore.dll]
std::invoke<void (__cdecl bs::CoreThread::*& __ptr64)(void) __ptr64,bs::CoreThread * __ptr64 & __ptr64>() - 0x140709846906864 File[type_traits:1445 (0)] Module[BansheeCore.dll]
std::_Invoke_ret<void (__cdecl bs::CoreThread::*& __ptr64)(void) __ptr64,bs::CoreThread * __ptr64 & __ptr64>() - 0x140709846882148 File[type_traits:1476 (0)] Module[BansheeCore.dll]
std::_Call_binder<std::_Unforced,0,void (__cdecl bs::CoreThread::*)(void) __ptr64,std::tuple<bs::CoreThread * __ptr64>,std::tuple<> >() - 0x140709846874798 File[functional:827 (0)] Module[BansheeCore.dll]
std::_Binder<std::_Unforced,void (__cdecl bs::CoreThread::*)(void) __ptr64,bs::CoreThread * __ptr64 const>::operator()<>() - 0x140709846873096 File[functional:881 (136)] Module[BansheeCore.dll]
std::_Invoker_functor::_Call<std::_Binder<std::_Unforced,void (__cdecl bs::CoreThread::*)(void) __ptr64,bs::CoreThread * __ptr64 const> & __ptr64>() - 0x140709846874126 File[type_traits:1377 (0)] Module[BansheeCore.dll]
std::invoke<std::_Binder<std::_Unforced,void (__cdecl bs::CoreThread::*)(void) __ptr64,bs::CoreThread * __ptr64 const> & __ptr64>() - 0x140709846907118 File[type_traits:1445 (0)] Module[BansheeCore.dll]
std::_Invoke_ret<void,std::_Binder<std::_Unforced,void (__cdecl bs::CoreThread::*)(void) __ptr64,bs::CoreThread * __ptr64 const> & __ptr64>() - 0x140709846882402 File[type_traits:1462 (0)] Module[BansheeCore.dll]
std::_Func_impl<std::_Binder<std::_Unforced,void (__cdecl bs::CoreThread::*)(void) __ptr64,bs::CoreThread * __ptr64 const>,std::allocator<int>,void>::_Do_call() - 0x140709846974213 File[functional:214 (0)] Module[BansheeCore.dll]
std::_Func_class<void>::operator()() - 0x140710191899706 File[functional:280 (0)] Module[BansheeUtility.dll]
bs::PooledThread::run() - 0x140710192418359 File[bsthreadpool.cpp:125 (0)] Module[BansheeUtility.dll]
std::_Invoker_pmf_pointer::_Call<void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 & __ptr64>() - 0x140710192378340 File[type_traits:1342 (0)] Module[BansheeUtility.dll]
std::invoke<void (__cdecl bs::PooledThread::*& __ptr64)(void) __ptr64,bs::PooledThread * __ptr64 & __ptr64>() - 0x140710192390464 File[type_traits:1445 (0)] Module[BansheeUtility.dll]
std::_Invoke_ret<void (__cdecl bs::PooledThread::*& __ptr64)(void) __ptr64,bs::PooledThread * __ptr64 & __ptr64>() - 0x140710192383188 File[type_traits:1476 (0)] Module[BansheeUtility.dll]
std::_Call_binder<std::_Unforced,0,void (__cdecl bs::PooledThread::*)(void) __ptr64,std::tuple<bs::PooledThread * __ptr64>,std::tuple<> >() - 0x140710192378606 File[functional:827 (0)] Module[BansheeUtility.dll]
std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const>::operator()<>() - 0x140710192378216 File[functional:881 (136)] Module[BansheeUtility.dll]
std::_Invoker_functor::_Call<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> >() - 0x140710192378446 File[type_traits:1377 (0)] Module[BansheeUtility.dll]
std::invoke<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> >() - 0x140710192390574 File[type_traits:1445 (0)] Module[BansheeUtility.dll]
std::_LaunchPad<std::unique_ptr<std::tuple<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> >,std::default_delete<std::tuple<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> > > > >::_Execute<0>() - 0x140710192380970 File[xthread:241 (0)] Module[BansheeUtility.dll]
std::_LaunchPad<std::unique_ptr<std::tuple<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> >,std::default_delete<std::tuple<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> > > > >::_Run() - 0x140710192407293 File[xthread:250 (0)] Module[BansheeUtility.dll]
std::_LaunchPad<std::unique_ptr<std::tuple<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> >,std::default_delete<std::tuple<std::_Binder<std::_Unforced,void (__cdecl bs::PooledThread::*)(void) __ptr64,bs::PooledThread * __ptr64 const> > > > >::_Go() - 0x140710192404022 File[xthread:233 (0)] Module[BansheeUtility.dll]
std::_Pad::_Call_func() - 0x140710192401933 File[xthread:210 (0)] Module[BansheeUtility.dll]
0x140710275100840 Module[ucrtbased.dll]
0x140710275099921 Module[ucrtbased.dll]
0x140710963650532 Module[KERNEL32.DLL]
0x140711001059217 Module[ntdll.dll]
Le thread 0x12e4 s'est arrêté avec le code 0 (0x0).

```

**Comment by BearishSun** [comment]
Resolved with 58b6cdd959b0753dac688508caa870b6a08a4300.

**Comment by christianclavet** [comment]
Confirmed, there is no crash anymore when the camera view is selected. 

Remark:
The camera render is black and nothing is displayed until we change the aspect ratio drop down menu (free) to another item. So the camera aspect ratio value is probably undefined when the application is not initialized.

**Comment by christianclavet** [comment]
Fixed the problem with the camera at startup. I added this line in GameWindow.cs at the end of the _OnInitialize() method:_
` OnAspectRatioChanged(0);`
This will simply tell Banshee that the aspect ratio as changed to the free aspect ratio (default at initialization) and will set up accordingly. This was not set at initialization.

**Comment by BearishSun** [comment]
Thanks. Will check it out. I tried today but havent managed to reproduce
it, but havent tried in opengl and release mode yet.

Dana 24. pro 2017. 20:00 osoba "Christian Clavet" <notifications@github.com>
napisala je:

Fixed the problem with the camera at startup. I added this line in
GameWindow.cs at the end of the *OnInitialize() method:*
OnAspectRatioChanged(0);
This will simply tell Banshee that the aspect ratio as changed to the free
aspect ratio (default at initialization) and will set up accordingly. This
was not set at initialization.

—
You are receiving this because you modified the open/close state.

Reply to this email directly, view it on GitHub
<https://github.com/BearishSun/BansheeEngine/issues/117#issuecomment-353798870>,
or mute the thread
